src/sdk/pynni/nni/nas/pytorch/classic_nas/mutator.py

```diff
@@ -0,0 +1,187 @@
+import os
+import sys
+import json
+import logging
+import torch
+import nni
+from nni.env_vars import trial_env_vars
+from nni.nas.pytorch.base_mutator import BaseMutator
+from nni.nas.pytorch.mutables import LayerChoice, InputChoice
+
+logger = logging.getLogger(__name__)
+
+def get_apply_next_architecture(model):
+    """
+    Wrapper of ClassicMutator to make it more meaningful,
+    similar to ```get_next_parameter``` for HPO.
+
+    Parameters
+    ----------
+    model : pytorch model
+        user's model with search space (e.g., LayerChoice, InputChoice) embedded in it
+    """
+    ClassicMutator(model)
+
+class ClassicMutator(BaseMutator):
+    """
+    This mutator is to apply the architecture chosen from tuner.
+    It implements the forward function of LayerChoice and InputChoice,
+    to only activate the chosen ones
+    """
+    def __init__(self, model):
+        """
+        Generate search space based on ```model```.
+        If env ```NNI_GEN_SEARCH_SPACE``` exists, this is in dry run mode for
+        generating search space for the experiment.
+        If not, there are still two mode, one is nni experiment mode where users
+        use ```nnictl``` to start an experiment. The other is standalone mode
+        where users directly run the trial command, this mode chooses the first
+        one(s) for each LayerChoice and InputChoice.
+
+        Parameters
+        ----------
+        model : pytorch model
+            user's model with search space (e.g., LayerChoice, InputChoice) embedded in it
+        """
+        super(ClassicMutator, self).__init__(model)
+        self.chosen_arch = {}
+        self.search_space = self._generate_search_space()
+        if os.environ.get('NNI_GEN_SEARCH_SPACE') is not None:
+            # dry run for only generating search space
+            self._dump_search_space(self.search_space)
+            sys.exit(0)
+        # get chosen arch from tuner
+        self.chosen_arch = nni.get_next_parameter()
+        if not self.chosen_arch and trial_env_vars.NNI_PLATFORM is None:
+            logger.warning('This is in standalone mode, the chosen are the first one(s)')
+            self.chosen_arch = self._standalone_generate_chosen()
+        self._validate_chosen_arch()
+
+    def _validate_chosen_arch(self):
+        pass
+
+    def _standalone_generate_chosen(self):
+        """
+        Generate the chosen architecture for standalone mode,
+        i.e., choose the first one(s) for LayerChoice and InputChoice
+
+        Returns
+        -------
+        dict
+            the chosen architecture
+        """
+        chosen_arch = {}
```

Comment:
Is this format here same as the format got from `get_next_parameter`? because I don't see any `_idx` here. And the intended question is, if we don't use `_idx`, what if I have two `NO_KEY` in InputChoice, in which case I can't distinguish them from values?

Reply:
good catch, forgot to update this part, thanks!